web/src/user_profile.ts

```diff
@@ -301,12 +302,16 @@ function format_user_stream_list_item_html(stream: StreamSubscription, user: Use
     });
 }
 
-function format_user_group_list_item_html(group: UserGroup): string {
+function format_user_group_list_item_html(group: UserGroup, user: User): string {
+    const is_me = user.user_id === current_user.user_id;
+    const is_direct_member = group.members.has(user.user_id);
     return render_user_group_list_item({
         group_id: group.id,
         name: group.name,
         group_edit_url: hash_util.group_edit_url(group, "general"),
         is_guest: current_user.is_guest,
+        is_direct_member,
+        can_remove_subscribers: settings_data.can_manage_user_group(group.id) || is_me,
```

Comment:
There can also be a case when user themselves cannot remove them from group if they are not allowed by `can_leave_group` setting.